component/component_test.js

```diff
@@ -1558,6 +1558,31 @@ steal("can", "can/map/define", "can/component", "can/view/stache" ,"can/route",
 
 	});
 
+	test("attach events on init", function(){
+
+		expect(2);
+		can.Component.extend({
+			tag: 'app-foo',
+			template: can.stache('<div>click me</div>'),
+			events: {
+				init: function(){
+					this.on("div", 'click', 'doSomethingfromInit');
+				},
+				inserted: function(){
+					this.on("div", 'click', 'doSomethingfromInserted');
+				},
+				doSomethingfromInserted: function(){
+					ok(true, "bound in inserted");
+				},
+				doSomethingfromInit: function(){
+					ok(true, "bound in init");
+				}
+			}
+		});
+		can.append( can.$("#qunit-fixture"), can.stache("<app-foo></app-foo>")({}));
+		can.$("#qunit-fixture div").trigger('click');
```

Comment:
This needs to be [can.trigger(el, ev)](http://canjs.com/docs/can.trigger.html) then tests should pass with the other libraries as well.